test/e2e/set/scyllacluster/scyllacluster_cleanup.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2023 ScyllaDB.
+
+package scyllacluster
+
+import (
+	"context"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	"github.com/scylladb/scylla-operator/pkg/helpers"
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	scyllafixture "github.com/scylladb/scylla-operator/test/e2e/fixture/scylla"
+	"github.com/scylladb/scylla-operator/test/e2e/framework"
+	"github.com/scylladb/scylla-operator/test/e2e/utils"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+var _ = g.Describe("ScyllaCluster", func() {
+	defer g.GinkgoRecover()
+
+	f := framework.NewFramework("scyllacluster")
+
+	g.It("nodes are cleaned up after horizontal scaling", func() {
+		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+		defer cancel()
+
+		sc := scyllafixture.BasicScyllaCluster.ReadOrFail()
+		sc.Spec.Datacenter.Racks[0].Members = 1
+
+		jobListWatcher := &cache.ListWatch{
+			ListFunc: helpers.UncachedListFunc(func(options metav1.ListOptions) (runtime.Object, error) {
+				return f.KubeClient().BatchV1().Jobs(f.Namespace()).List(ctx, options)
+			}),
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				return f.KubeClient().BatchV1().Jobs(f.Namespace()).Watch(ctx, options)
+			},
+		}
+
+		framework.By("Creating a single node ScyllaCluster")
+
+		jobObserver := utils.ObserveObjects[*batchv1.Job](jobListWatcher)
+		jobObserver.Start(ctx)
```

Comment:
won't this be racy? watches can be arbitrarily delayed

Reply:
I assume we could setup the untilWithSync to run in parallel, write to a channel and the do should eventually. for the negative case we'd likely just do not see the object within /  sleep.

Reply:
Added explicit wait until inner informer is synced